### DISCUSSION TITLE
Linux Signature Problems - VAC Signature Verification, Linux Proton w/ Steam Tinker

### DISCUSSION THREAD

**Discussion by ghost** [cafali/SnapKey#36] [Linux]
Some of your game files have been detected to have no signatures or invalid signatures. You will not be allowed to join VAC secure servers. Please verify your launch options, check correctness of your game installation, restart the game and try again

I am using Steam Tinker to launch SnapKey.exe

Does anyone have a fix?

**Comment by cafali** [comment]
Have you tried the steps in the guide for running SnapKey on Linux yet?

Here’s the link: https://github.com/cafali/SnapKey/issues/4

**Reply by ghost** [reply]
yeah its installed succesfully. it works on local maps perfectly, i dont know why VAC is detecting it though

**Reply by cafali** [reply]
I’m not sure what’s going on, but everything’s running fine on Windows for me. Since you need to install and run more things on Linux to even run SnapKey, it might not be SnapKey itself but possibly the wrappers or other stuff you’re using. Maybe another Linux user can jump in and help if they see this. Sorry I can’t be of more help...

Also, why should SnapKey be detected by VAC? It doesn’t change any files or interact with the game directly. The error message says clearly: 'Please verify your launch options, check correctness of your game installation, restart the game and try again.'

This was also a common VAC error back in CSGO, and it was pretty random.

**Reply by minteeaa** [reply]
could be the interaction of SteamTinker within local files, as technically Tinker is a wrapper-for-a-wrapper in relation to Wine, emulating a windows system/drivers on top of the base linux system, so interactions with games and specific anticheats cannot be guaranteed or deterministic

VAC specifically can be finnicky on this, and the best option for prevention of any VAC bans would be a native linux solution

unfortunately, i'm finding key handling on X11 is a major pain; even worse on Wayland that doesn't even have proper support for anything that grabs inputs

definitely possible to get a linux-native build out, but be careful with Steam Tinker on games with particularly aggressive anticheats
